libs/datamodel/core/src/ast/parser/datamodel.pest

```diff
@@ -30,14 +30,18 @@ field_declaration = { doc_comment_and_new_line* ~ non_empty_identifier ~ LEGACY_
 // ######################################
 
 // Pest is greedy, order is very important here.
-field_type = { unsupported_optional_list_type | list_type | optional_type | legacy_required_type | legacy_list_type | base_type }
+field_type = { optional_unsupported_type | list_unsupported_type | unsupported_type | unsupported_optional_list_type | list_type | optional_type | legacy_required_type | legacy_list_type | base_type  }
 
-base_type = { non_empty_identifier } // Called base type to not conflict with type rust keyword
+//todo can these be merged?
+optional_unsupported_type = { "Unsupported(" ~ string_literal ~ ")" ~"?" }
+list_unsupported_type = { "Unsupported(" ~ string_literal ~ ")" ~"[]" }
+unsupported_type = { "Unsupported(" ~ string_literal ~ ")" }
+unsupported_optional_list_type = { non_empty_identifier ~ "[]" ~ "?" }
```

Comment:
Comment from the review call: maybe we should add  a comment saying it's not unsupported in the same sense. This line is syntax we don't support.